src/client/components/pages/parts/relationship-types-tree.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2023 Shivam Awasthi
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+import React, {useCallback, useState} from 'react';
+import {Button} from 'react-bootstrap';
+import {RelationshipTypeDataT} from '../../forms/type-editor/typeUtils';
+
+
+type RelationshipTypeTreePropsT = {
+	relationshipTypes: RelationshipTypeDataT[],
+	parentId?: number | null,
+	indentLevel?: number
+};
+
+function RelationshipTypeTree({relationshipTypes, parentId, indentLevel}: RelationshipTypeTreePropsT) {
+	const [expandedRelationshipTypeIds, setExpandedRelationshipTypeIds] = useState([]);
+
+	function toggleExpand(relTypeId) {
```

Comment:
As a matter of fact, I did try Accordion. It does not look clean with the tree structure(maybe there are ways to make it work, but I was not able to make it into a nice looking tree structure). Had it been for something where the levels of all the sub-components are same, then it would have worked.